chainercv/links/ssd/ssd.py

```diff
@@ -0,0 +1,257 @@
+from __future__ import division
+
+import itertools
+import numpy as np
+import six
+
+import chainer
+import chainer.functions as F
+from chainer import initializers
+import chainer.links as L
+
+from chainercv import transforms
+
+
+class SSD(chainer.Chain):
+    """Base class of Single Shot Multibox Detector [1].
+
+    This is a base class of Single Shot Multibox Detector.
+    It requires a feature extraction method and a preprocessing method.
+
+    [1] Wei Liu, Dragomir Anguelov, Dumitru Erhan, Christian Szegedy,
+    Scott Reed, Cheng-Yang Fu, Alexander C. Berg.
+    SSD: Single Shot MultiBox Detector. ECCV 2016.
+
+    Args:
+        n_class (int): The number of classes.
+        links (dict of chainer.Link): A dictionary of :class:`chainer.Link`.
+            The links  will added to the chain as child links.
+    Parameters:
+        grids (iterable of int): The sizes of feature maps.
+        aspect_ratios (iterable of tuple or int): The aspect ratios of
+            default bounding boxes for each feature map.
+        steps (iterable of float): The step size for each feature map.
+        sizes (iterable of float): The base size of default bounding boxes
+            for each feature map.
+        variance (tuple of float): Two coefficients for encoding
+            the locations of bounding boxe. The first value is used to
+            encode coordinates of the centers. The second value is used to
+            encode the sizes of bounding boxes.
+            The default values are 0.1 and 0.2.
+        nms_threshold (float): The threshold value
+            for :meth:`chainercv.transfroms.non_maximum_suppression`.
+            The default value is 0.45.
+        score_threshold (float): The threshold value for confidence score.
+            If a bounding box whose confidence score is lower than this value,
+            the bounding box will be suppressed. The default value is 0.6.
+            This value is optimized for visualization.
+            For evaluation, the optimized value is 0.01.
+    """
+
+    variance = (0.1, 0.2)
+
+    nms_threshold = 0.45
+    score_threshold = 0.6
+
+    conv_init = {
+        'initialW': initializers.GlorotUniform(),
+        'initial_bias': initializers.Zero(),
+    }
```

Comment:
In my understanding, mutable variables should not be declared at class level because they are constants for class and should not vary among class instantiations.
Instead, it should be declared inside `__init__` regardless of taking it as an argument. Since, any public attributes are public in Python, this will let users to modify it whenever needed.

On giving paramters as arguments to `__init__` or not, I think either way is fine.